### PR TITLE
Webapp bugs

### PR DESCRIPTION
**Pull request by borkopetrovicc**
## Description

https://www.notion.so/jwplayer/Webapp-bugs-51c77f5be0b34cdc9789b88bf597f0a6

### Steps completed:

According to our definition of done, I have completed the following steps:

- [ ] Acceptance criteria met
- [ ] Unit tests added
- [ ] Docs updated (including config and env variables)
- [ ] Translations added
- [ ] UX tested
- [ ] Browsers / platforms tested
- [ ] Rebased & ready to merge without conflicts
- [ ] Reviewed own code

### CODE REVIEW

File: src/containers/AccountModal/forms/EditPassword.tsx [hunk 1]
```diff
@@ -40,8 +41,8 @@ const ResetPassword: React.FC = () => {
         }
         await changePasswordWithToken(emailParam || '', password, resetToken, passwordConfirmation);
       }
-      await logout();
-      navigate(addQueryParams(window.location.origin, { u: 'login' }));
+      await logout(true);
```
Comment (budzworthy) [inline src/containers/AccountModal/forms/EditPassword.tsx]: Why do we not actually log the user out here? Have you tested this change with the Cleeng integration?

File: src/stores/AccountController.ts [hunk 2]
```diff
@@ -146,7 +146,7 @@ export const login = async (email: string, password: string) => {
     useAccountStore.setState({ loading: false });
   });
 };
-export async function logout() {
+export async function logout(onlyState = false) {
```
Comment (budzworthy) [inline src/stores/AccountController.ts]: I would suggest splitting out the `clearLoginState` logic to its own method if it turns out we need it. An optional flag is kind of hard to intuit, especially when the logic is inverted where `logout(false)` means to actually logout and `logout(true)` means not to.

File: src/containers/AccountModal/forms/EditPassword.tsx (hunk 1 above)
Comment (borkopetrovicc) [reply src/containers/AccountModal/forms/EditPassword.tsx]: 075a5d6bde7bc10485a74af46852c40e0e35bf9b

File: src/stores/AccountController.ts (hunk 2 above)
Comment (borkopetrovicc) [reply src/stores/AccountController.ts]: 075a5d6bde7bc10485a74af46852c40e0e35bf9b

**Review by kmitrovv — APPROVED** [review]

File: src/stores/AccountController.ts [hunk 3]
```diff
@@ -146,26 +146,36 @@ export const login = async (email: string, password: string) => {
...
+}
+
+type LogoutOptions = {
+  includeNetworkRequest: boolean;
+};
 
-    await restoreFavorites();
-    await restoreWatchHistory();
-    await accountService?.logout();
+export async function logout(logoutOptions: LogoutOptions = { includeNetworkRequest: true }) {
```
Comment (kmitrovv) [inline src/stores/AccountController.ts]: You can delete the type defined above not really needed for just on param
```suggestion
export async function logout(logoutOptions: { includeNetworkRequest: boolean }) {
```

Comment (kiremitrov123) [inline src/stores/AccountController.ts:169]:
```suggestion
```

**Pull request merged**